### PR TITLE
MAT-7855 Integrate HAPI FHIR Server into Test Case Builder for internal value set expansion

### PR DESCRIPTION
**Pull request by adongare**
## Terminology Service PR

Jira Ticket: [MAT-7855](https://jira.cms.gov/browse/MAT-7855)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/InternalTerminologyService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    try {
+      log.info("Fetching ValueSet expansion for {}", id);
+      Parameters parameters =
+          hapiClient
+              .operation()
+              .onInstance(new IdType("ValueSet", id))
+              .named("$expand")
+              .withNoParameters(Parameters.class)
+              .execute();
+      return (ValueSet) parameters.getParameter().get(0).getResource();
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/InternalTerminologyService.java:38]: Is there a chance of HAPI returning something other than a ValueSet Resource if it errors, or will it only throw an Exception?

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/gov/cms/madie/terminology/service/InternalTerminologyService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/terminology/service/InternalTerminologyService.java:38]: because I'm specifying the instance on which the operation should be performed, I'll always get value set resource if it was successful operation. For errors it only throws reception. we don't get the resource in that case.

**Review by jkotanchik-SB — APPROVED** [review]
